src/runtime/SmallValue.h

```diff
@@ -215,32 +205,15 @@ class SmallValue {
 
     bool isEmpty() const
     {
-        if (HAS_OBJECT_TAG(m_data.payload)) {
-            PointerValue* v = (PointerValue*)m_data.payload;
-            if (((size_t)v) == smallValueEmpty) {
-                return true;
-            }
-        }
-        return false;
+        return m_data.payload == ValueEmpty;
     }
 
     operator Value() const
     {
         if (HAS_OBJECT_TAG(m_data.payload)) {
             PointerValue* v = (PointerValue*)m_data.payload;
```

Comment:
It would be better to use `HAS_SMI_TAG` instead of `HAS_OBJECT_TAG' here, because this if statement checks if this SmallValue has a smi value or not.

Reply:
I am not sure I follow. Isn't using `HAS_SMI_TAG` inverts the condition? The check is coming from the old code (and I don't really understand its purpose, just what it does). If you have a better suggestion please give me a code snippet.

Reply:
`HAS_SMI_TAG` checks whether the least significant bit is 1 or not. Code at this point just checks if value has a smi,  and it convert the smi value to integer according to that check result,
`HAS_OBJECT_TAG` in here is somewhat confusing and it would be more proper to use `HAS_SMI_TAG`. However its just my suggestion and you can consider this in later patches. Thank you.